src/controls/first_person.rs

```diff
@@ -3,25 +3,25 @@ use mint;
 use std::ops;
 
 use cgmath::Rotation3;
-use input::{Input, Key, KeyAxis};
+use input::{Axis, Input, Key};
 use object::Object;
 use std::f32::consts::PI;
 
 #[derive(Clone, Debug)]
 struct Axes {
-    pub forward: Option<KeyAxis>,
-    pub strafing: Option<KeyAxis>,
-    pub vertical: Option<KeyAxis>,
+    pub forward: Option<Axis>,
```

Comment:
The reason I made `KeyAxis` a separate type as opposed to a more generic `Axis` is because with keys it's discreet, while mouse/joystick axis is continuous. This semantics difference makes for different APIs making sense, thus - different types.

Reply:
Perhaps we should rename to make this distinction clear, i.e. `Discrete` and `Continuous` axes.

Reply:
I thought we've reached consensus... I'm not sure. I need to check one thing for possibility of implementation before final decision...

Reply:
> Accessing input data with indexing (like "input.pressed()[button]") is impossible due to current implementation of ops::Index and borrow checker

Is it, really? Please provide what you got. I think `input[button].is_hit()` would be quite possible, assuming `button` is `Copy`, and `input` stores actual structures per key that it can return as references.

Reply:
Well, your syntax (`input[button].is_hit()`) is possible, yes. Impossible one is `input.pressed()[button]`

Reply:
The former is more intuitive.